internal/types/array.go

```diff
@@ -111,3 +111,38 @@ func (a *Array) Append(values ...any) error {
 func (a *Array) RemoveByPath(path Path) {
 	removeByPath(a, path)
 }
+
+// Contains checks if the Array contains the given value.
+func (a *Array) Contains(value any) bool {
+	for _, v := range a.s {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsAll checks if the Array contains all the given values of the Array b.
+// Currently, this algorithm is O(n^2) without any performance tuning.
+// Possible improvement: to sorts both arrays before comparing.
+// If sorting is implemented, calling of the Contains function will be O(n*log(n)) instead of O(n^2).
+func (a *Array) ContainsAll(b *Array) bool {
+	if a.Len() < b.Len() {
+		// TODO: How to validate that all the values in b are unique?
+		return false
+	}
+
+	// TODO: This place can be significantly improved:
```

Comment:
To measure effectiveness, it's necessary to add other tests, loaded with data including big arrays and the number of arrays will be also big enough to do a high load-testing.

It's a separate task and definitely, a separate PR and it can become a larger task and cover not only this one `$all` operation but also other update operators. And it can be measured carefully the influence on other operators performance... So it's a big another task.

Reply:
@seeforschauer well, I don't need to add other tests to tell that O(n^2) is worse than O(n*log n) :)

Reply:
> @seeforschauer well, I don't need to add other tests to tell that `O(n^2)` is worse than `O(n*log n)` :)

Then why not just implement it in a better way? What is needed for it?

Reply:
@seeforschauer time :) Basically, my main question is about the future strategy. If all the things like `$all` should be called on the backend level, it doesn't make sense to use fancy algorithms on FerretDB. But if we are going to leave with `$all` being a "post-filter", I'd rather invest some time to offer a better algorithm right now.